### PR TITLE
[4.0] Banners - nullable datetime fields

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #16788 ....same as #24675 for contacts - I really just copied the work of others (someone has to)

### Summary of Changes
some datetime fields should be nullable as default

- publish_up
- publish_down
- checked_out_time

### Testing Instructions
test com_banners

### Expected result
banners should works as before

### CODE REVIEW

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -141,9 +141,9 @@ CREATE TABLE IF NOT EXISTS `#__banners` (
   `track_clicks` tinyint(4) NOT NULL DEFAULT -1,
   `track_impressions` tinyint(4) NOT NULL DEFAULT -1,
   `checked_out` int(10) unsigned NOT NULL DEFAULT 0,
-  `checked_out_time` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-  `publish_up` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-  `publish_down` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
+  `checked_out_time` datetime,
+  `publish_up` datetime,
+  `publish_down` datetime,
   `reset` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql:147]: `reset` can be null?

**Comment by ghost** [comment]
@brianteeman i guess #16788 cannot be closed for now.

**Comment by brianteeman** [comment]
No not yet there is still more components

File: installation/sql/mysql/joomla.sql (hunk 1 above)
Comment (brianteeman) [reply installation/sql/mysql/joomla.sql:147]: I have just copied from the example. So I don't know

Comment (richard67) [reply installation/sql/mysql/joomla.sql:147]: @alikon In your PR #24675 , which is the template for this here, you handled only columns checked_out_time, publish_up and publish_down. What do you think, should Brian continue like that, or should he handle also other columns which should be nullable? From a functional point of view, the reset column here should be nullable, too, for the case if there never has been a reset. On the other hand, if in all other PRs right now only the 3 columns mentioned above are handled, this PR here would be consistent with it, and we would know we have to check for other columns later again for all components. My opinion is that it should be consistent, so this PR is ok for me now. But I'd like to know your opinion on it, too.

**Comment by richard67** [comment]
@brianteeman There is also a column `checked_out_time` in the `#__banner_clients` table. Shouldn't that be done with this PR, too?

File: installation/sql/mysql/joomla.sql (hunk 1 above)
Comment (alikon) [reply installation/sql/mysql/joomla.sql:147]: well, different tables have different fields/behaviour, that said, following this comment as only an example https://github.com/joomla/joomla-cms/pull/24675#issuecomment-485391598   and without a proper effort to try to move forwrad joomla db schema from mambo times ....... (sorry for the rant)....for me too the reset field should be nullable......but again  without a full re-think about db schema ......

p.s
thanks Brian

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**